core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -190,6 +193,8 @@ private[kafka] abstract class AbstractServerThread(connectionQuotas: ConnectionQ
    */
   protected def startupComplete() = {
     startupLatch.countDown()
+    // Activate the shutdown latch by resetting it to count 1
+    shutdownLatch = new CountDownLatch(1)
```

Comment:
I think these two latches have no impact with each other. One is for startup and one is used for shutdown. Of course, I could put it before startupLatch.countDown.